### ISSUE TITLE
&c. Abbreviation

### ISSUE THREAD

**Issue by gasyoun** [sanskrit-lexicon/MWS#86]
`&c.` is an abbreviation and not an obvious one.

![vatsa](https://user-images.githubusercontent.com/80761/103436729-fdca4e80-4c2f-11eb-9aaf-ad6c80d0160b.JPG)

`etcetera`?

**Comment by Andhrabharati** [comment]
& is a symbolic represenation of Latin 'et' ( English 'and'), similar to ॐ being a symbolic represention for Sanskrit 'om'.

And it is noticed that at many places in the data this '&' is used instead of 'and' (which is the way the book has). All such need to be brought back to the 'book form' of 'and'.

&c. is Latin 'et cetera', its English equivalent being 'and others'.

**Comment by Andhrabharati** [comment]
This and other such composite words in Latin (like i.e., e.g., ...) could be expanded with both Latin forms and English forms, the present work being for English knowing people, who may or may not know Latin.

It may be noted that all the (standard) present day English dictionaries give the expansion in English, for all such abbreviations.

**Comment by funderburkjim** [comment]
> at many places in the data this '&' is used instead of 'and' (which is the way the book has). All such need to be brought back to the 'book form' of 'and'.

Agree.

**Comment by funderburkjim** [comment]
> &c. is an abbreviation and not an obvious one

Had not realized it is not obvious.  It's obviousness to me must be due to English as my home language.

We can add the `<ab>&c.</ab>` markup sometime,  and then the expansion 
`et cetera (Latin); X`

Where X could be 'and others' as AB suggests, or one of several other possibilities -- Not sure which is best.
```
synonyms:
and so on · 
and so forth · 
and so on and so forth · 
and the rest · 
and/or the like · 
and/or suchlike · 
and/or more of the same · 
and/or similar things · 
et cetera et cetera · 
and others · 
among others · 
et al. · 
and what have you · 
and whatnot
```

**Comment by gasyoun** [comment]
> Had not realized it is not obvious. It's obviousness to me must be due to English as my home language.

”The world is full of obvious things which nobody ever observes,” says Sherlock Holmes

> We can add the <ab>&c.</ab> markup sometime

Let it be now.

> Not sure which is best.

Neither do I. So let's do what's clear.

**Comment by Andhrabharati** [comment]
I think `and so on` is the better choice; `and others` to go with `et al.` (suits more than here).

**Comment by gasyoun** [comment]
@funderburkjim as an unmarked abbreviation it is too big compare to other abbreviations.

![fdssdffdsfds](https://user-images.githubusercontent.com/80761/115139322-ede94580-a039-11eb-9bb2-983904544f84.png)

**Comment by funderburkjim** [comment]
I agree that '&c.' is so common that it seems undesirable to add markup such as `<ab>&c.</ab>` throughout mw.txt.  I suppose the displays could add tooltips  during html generation.  

@Andhrabharati may find this suggestion insufficient.  For instance if there are cases where the Cologne digitization has substituted '&c.'  for the printed text 'and',  then these differences would need to be changed in mw.txt.

**Comment by Andhrabharati** [comment]
No, the printed word 'and' is typed just as &, not as &c. (so far as I saw)

**Comment by gasyoun** [comment]
> may find this suggestion insufficient

As you see he is fine. I'm not fine leaving as it is now. It is way too big now - because undermarked.

**Comment by Andhrabharati** [comment]
Probably  the &c. could be dealt the same way as etc. is dealt. (both  being the same)

**Comment by gasyoun** [comment]
<details>
<summary>🤖 Bug triage findings</summary>

**Bug class:** markup (missing `<ab>` tags on a ubiquitous abbreviation)

**Evidence:**
- `&c.` appears **21,791 times** in `csl-orig/v02/mw/mw.txt` — all unmarkd-up.
- `<ab>&c.</ab>` occurrences: **0** (none yet wrapped).
- For comparison, `etc.` (same meaning) has 6 occurrences and is already fully wrapped as `<ab>etc.</ab>`.
- Pattern used: `Select-String -Pattern "&c\."` on the local clone of csl-orig.

**Agent-fixable (Tier A)?** Mechanically yes — a bulk `s/&c\./<ab>&c.<\/ab>/g` across all 21,791 lines — but the thread shows an unresolved editorial decision: whether to mark up at the source level or handle via tooltip/HTML-generation in the display layer. Jim's 2021-04-23 comment explicitly proposed the display-layer route instead of source markup.

**Proposed fix (if Tier A):**
If the decision is made to mark up at source level, the change is straightforward: replace every bare `&c.` with `<ab>&c.</ab>` in mw.txt (21,791 lines). XML validation after application is mandatory.

If the display-layer route is preferred instead, no source change is needed; the generation scripts in csl-pywork would add a tooltip for `&c.` the same way they handle other recognized abbreviations.

**If Tier D — what needs checking:**
If the display-layer route is chosen, check csl-pywork and csl-websanlexicon for where `<ab>` expansions are rendered, and add `&c.` to that lookup.

**Suggested next step:** Resolve the open editorial decision: source-markup (`<ab>&c.</ab>` batch in mw.txt) vs. display-layer tooltip. Once decided, the bulk change is straightforward in either direction.

</details>

**Comment by drdhaval2785** [comment]
It is OK to add ab tag in mw.txt file itself. Every abbreviation is handled thus.

**Comment by gasyoun** [comment]
**H1523 display-layer fix (PRs open)**

Per Jim (display-layer tooltip OK) and Dhaval (source `<ab>` also OK): this pass lands the **display-layer** path so ~21k bare `&c.` get hover **et cetera; and so on** without a 21k-line csl-orig batch.

- https://github.com/sanskrit-lexicon/csl-websanlexicon/pull/91
- https://github.com/sanskrit-lexicon/csl-apidev/pull/87

Optional follow-up (monthly csl-orig batch): bulk wrap `&c.` → `<ab>&c.</ab>` + add `&c.` to mwab with the same expansion.

**Comment by gasyoun** [comment]
**H1523 hygiene close — display-layer fix shipped**

Bare `&c.` now gets hover tooltip **et cetera; and so on** via `basicadjust.php` (no 21k-line source rewrite required for the user-visible gap).

Merged:
- https://github.com/sanskrit-lexicon/csl-websanlexicon/pull/91
- https://github.com/sanskrit-lexicon/csl-apidev/pull/87

Optional follow-up (not blocking close): bulk `<ab>&c.</ab>` in `mw.txt` via monthly csl-orig batch per Dhaval 2026-06-28. Server deploy of the display templates is the remaining operator step for production.
